gst/nnstreamer/tensor_filter/tensor_filter_support_cc.hh

```diff
@@ -0,0 +1,126 @@
+
+    static const GstTensorFilterFramework fwdesc_template; /**< Template for fwdesc. Each subclass or object may have different subplugin_data while the C callbacks are identical. Thus, we copy C callbacks from this template to fwdesc and let each subplugin start customizing based on fwdesc, not on fwdesc_template */
+
+    /** tensor_filter/C wrapper functions */
+    static int cpp_open (const GstTensorFilterProperties * prop, void **private_data); /**< C wrapper func, open */
+    static void cpp_close (const GstTensorFilterProperties * prop, void **private_data); /**< C wrapper func, close */
+    static int cpp_invoke (const GstTensorFilterProperties *prop, void *private_data, const GstTensorMemory *input, GstTensorMemory *output); /**< C V1 wrapper func, invoke */
+    static int cpp_getFrameworkInfo (const GstTensorFilterProperties * prop, void *private_data, GstTensorFilterFrameworkInfo *fw_info); /**< C V1 wrapper func, getFrameworkInfo */
+    static int cpp_getModelInfo (const GstTensorFilterProperties * prop, void *private_data, model_info_ops ops, GstTensorsInfo *in_info, GstTensorsInfo *out_info); /**< C V1 wrapper func, getModelInfo */
+    static int cpp_eventHandler (const GstTensorFilterProperties * prop, void *private_data, event_ops ops, GstTensorFilterFrameworkEventData *data); /**< C V1 wrapper func, eventHandler */
+
+    GstTensorFilterFramework fwdesc; /**< Represents C/V1 wrapper for the derived class and its objects. Derived should not access this anyway; the base class will handle this with the C wrapper functions, base static-functions, and base constructors/destructors. */
+
+  protected: /** Derived classes should call these at init/exit */
+    template<typename T>
+    static T * register_subplugin ();
+        /**< Register this subplugin class (not object) (usually at so-init)
+          * @retval: an empty instance for unregister_subplugin */
+
+    template<typename T>
+    static void unregister_subplugin (T * emptyInstance);
+        /**< Unregister this subplugin class (not object) (usually at so-exit) */
+
+  public:
+    /*************************************************************
+     ** These should be filled/implemented by subplugin authors **
+     *************************************************************/
+    tensor_filter_subplugin ();
+        /**< Creates a non-functional "empty" object
+             Subplugin (derived class) should make a constructor with same role and input arguments!
+          */
+
+    virtual tensor_filter_subplugin * getEmptyInstance () = 0;
+        /**< Do return derived_class(), the "empty" object */
+
+    virtual void configure_instance (const GstTensorFilterProperties *prop) = 0;
+        /**< Configure a non-functional "empty" object as a
+             functional "filled" object ready to be invoked */
+
+    virtual ~tensor_filter_subplugin (); /**< called by Close */
+
+    virtual int invoke (const GstTensorMemory * input, GstTensorMemory * output) = 0;
```

Comment:
Q: Is it possible to use references instead of pointers for the const parameters. If it is, we do not need to check whether a given const argument is nullptr or valid in the derived functions.

Reply:
Oh.. that's good. I've been too much C-oriented (for 30 years... it was the dominant programming language for me)

Reply:
Now, all the C++ virtual functions of the class use references instead of pointers for their parameters.